cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java

```diff
@@ -158,4 +164,13 @@ public void deleteTopic(final String topic) throws TopicNotFoundException, IOExc
       throw new RuntimeException(e);
     }
   }
+
+  @Override
+  public boolean namespaceExists(String namespace) throws IOException {
+    try {
+      return namespaceQueryAdmin.exists(new NamespaceId(namespace));
+    } catch (Exception e) {
```

Comment:
It's better to throw `IOException` as is if the exception `e` is already an `IOException`.

```
} catch (IOException e) {
  throw e;
} catch (Exception e) {
  throw new IOException(e);
}